cmake/ports/zlib/cmake_dont_build_more_than_needed.patch

```diff
@@ -10,32 +10,64 @@ index 0fe939d..8d2f5f1 100644
  
  set(INSTALL_BIN_DIR "${CMAKE_INSTALL_PREFIX}/bin" CACHE PATH "Installation directory for executables")
  set(INSTALL_LIB_DIR "${CMAKE_INSTALL_PREFIX}/lib" CACHE PATH "Installation directory for libraries")
-@@ -211,7 +212,15 @@ elseif(BUILD_SHARED_LIBS AND WIN32)
+@@ -124,9 +125,11 @@ set(ZLIB_SRCS
+ )
+ 
+ if(NOT MINGW)
+-    set(ZLIB_DLL_SRCS
+-        win32/zlib1.rc # If present will override custom build rule below.
+-    )
++    if(BUILD_SHARED_LIBS)
++        set(ZLIB_DLL_SRCS
++            win32/zlib1.rc # If present will override custom build rule below.
++        )
++    endif()
+ endif()
+ 
+ if(CMAKE_COMPILER_IS_GNUCC)
+@@ -180,11 +183,12 @@ if(MINGW)
+                             -I ${CMAKE_CURRENT_BINARY_DIR}
+                             -o ${CMAKE_CURRENT_BINARY_DIR}/zlib1rc.obj
+                             -i ${CMAKE_CURRENT_SOURCE_DIR}/win32/zlib1.rc)
+-    set(ZLIB_DLL_SRCS ${CMAKE_CURRENT_BINARY_DIR}/zlib1rc.obj)
++    if(BUILD_SHARED_LIBS)
++        set(ZLIB_DLL_SRCS ${CMAKE_CURRENT_BINARY_DIR}/zlib1rc.obj)
++    endif()
+ endif(MINGW)
+ 
+-add_library(zlib SHARED ${ZLIB_SRCS} ${ZLIB_ASMS} ${ZLIB_DLL_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
+-add_library(zlibstatic STATIC ${ZLIB_SRCS} ${ZLIB_ASMS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
++add_library(zlib ${ZLIB_SRCS} ${ZLIB_ASMS} ${ZLIB_DLL_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
+ set_target_properties(zlib PROPERTIES DEFINE_SYMBOL ZLIB_DLL)
+ set_target_properties(zlib PROPERTIES SOVERSION 1)
+ 
+@@ -201,7 +205,7 @@ endif()
+ 
+ if(UNIX)
+     # On unix-like platforms the library is almost always called libz
+-   set_target_properties(zlib zlibstatic PROPERTIES OUTPUT_NAME z)
++   set_target_properties(zlib PROPERTIES OUTPUT_NAME z)
+    if(NOT APPLE)
+      set_target_properties(zlib PROPERTIES LINK_FLAGS "-Wl,--version-script,\"${CMAKE_CURRENT_SOURCE_DIR}/zlib.map\"")
+    endif()
```

Comment:
Indentation.

Reply:
this is a patch from the vcpkg project with slight modifications. the patch would be applied to the zlib project which is basically an external lib, I don't know if we should make style changes here, it might be harder to update the patches if we do